templates/mroute_up-Debian.erb

```diff
@@ -7,7 +7,12 @@ if [ "$IFACE" = "<%= @interface -%>" ] || [ "$IFACE" = "--all" ]; then
 <% @routes.each do |net,gw| -%>
   if ! ip route show | grep -qF "<%= net %> "
   then
-    ip route add <%= net %> <% if /^\d/.match(gw) %>via<% else %>dev<% end %> <%= gw %>
+    ip route add <%= net %><%
+      if gw.kind_of?(Array)
+        gw.each do | g | %> nexthop via <%= g %><% end %>
+      <%- elsif /^\d/.match(gw) %> via <%= gw %>
```

Comment:
here and in the mroute_down template it looks like the generated ip route command is now on multiple lines  instead of the previous single line. Have you verified this? Does it work?

Reply:
Sure I did.
Space/line break trimming helps (https://puppet.com/docs/puppet/5.3/lang_template_erb.html#tags).

You can see real output.
via list:
```
root@i-9e4c2420:~# cat 1.pp
network::mroute {'eth1':
  routes => {
    "192.168.0.0/24" => ['10.70.112.1', '10.70.112.2'],
  }
}
root@i-9e4c2420:~# cat /etc/network/if-down.d/z90-route-eth1
#!/bin/bash
#
###
### File managed by Puppet
###
if [ "$IFACE" = "eth1" ] || [ "$IFACE" = "--all" ]; then
  if ip  route show  | grep -qF "192.168.0.0/24 "
  then
    ip route del 192.168.0.0/24 nexthop via 10.70.112.1 nexthop via 10.70.112.2
  fi
fi
root@i-9e4c2420:~# cat /etc/network/if-up.d/z90-route-eth1
#!/bin/bash
#
###
### File managed by Puppet
###
if [ "$IFACE" = "eth1" ] || [ "$IFACE" = "--all" ]; then
  if ! ip  route show  | grep -qF "192.168.0.0/24 "
  then
    ip route add 192.168.0.0/24 nexthop via 10.70.112.1 nexthop via 10.70.112.2
  fi
fi
```

via one ip:
```
root@i-9e4c2420:~# cat 1.pp
network::mroute {'eth1':
  routes => {
    "192.168.0.0/24" => '10.70.112.1',
  }
}
root@i-9e4c2420:~# cat /etc/network/if-up.d/z90-route-eth1
#!/bin/bash
#
###
### File managed by Puppet
###
if [ "$IFACE" = "eth1" ] || [ "$IFACE" = "--all" ]; then
  if ! ip  route show  | grep -qF "192.168.0.0/24 "
  then
    ip route add 192.168.0.0/24 via 10.70.112.1
  fi
fi
root@i-9e4c2420:~# cat /etc/network/if-down.d/z90-route-eth1
#!/bin/bash
#
###
### File managed by Puppet
###
if [ "$IFACE" = "eth1" ] || [ "$IFACE" = "--all" ]; then
  if ip  route show  | grep -qF "192.168.0.0/24 "
  then
    ip route del 192.168.0.0/24 via 10.70.112.1
  fi
fi
```

via device:
```
root@i-9e4c2420:~# cat 1.pp
network::mroute {'eth1':
  routes => {
    "192.168.0.0/24" => 'eth1',
  }
}
root@i-9e4c2420:~# cat /etc/network/if-up.d/z90-route-eth1
#!/bin/bash
#
###
### File managed by Puppet
###
if [ "$IFACE" = "eth1" ] || [ "$IFACE" = "--all" ]; then
  if ! ip  route show  | grep -qF "192.168.0.0/24 "
  then
    ip route add 192.168.0.0/24 dev eth1
  fi
fi
root@i-9e4c2420:~# cat /etc/network/if-down.d/z90-route-eth1
#!/bin/bash
#
###
### File managed by Puppet
###
if [ "$IFACE" = "eth1" ] || [ "$IFACE" = "--all" ]; then
  if ip  route show  | grep -qF "192.168.0.0/24 "
  then
    ip route del 192.168.0.0/24 dev eth1
  fi
fi
```

Same for CentOS/RedHat.